tests/unit/Store.ts

```diff
@@ -0,0 +1,251 @@
+		assert.equal(JSON.stringify(filteredCollection.queryLog), JSON.stringify(expectedQueryLog1));
+
+		filteredCollection = filteredCollection.filter(filter2);
+		assert.equal(JSON.stringify(filteredCollection.queryLog), JSON.stringify(expectedQueryLog2));
+	},
+
+	'sort'() {
+		const sortObject = { property: 'prop1', descending: true };
+		const sortObjectArray = [ sortObject, { property: 'prop2' } ];
+		const comparator = function comparator(a: any, b: any) {
+			return 0;
+		};
+		const expectedQueryLog1: any[] = [ {
+			type: 'sort',
+			arguments: [ sortObject.property, sortObject.descending ],
+			normalizedArguments: [ [ sortObject ] ]
+		} ];
+		const expectedQueryLog2: any[] = [ {
+			type: 'sort',
+			arguments: [ sortObject ],
+			normalizedArguments: [ [ sortObject ] ]
+		} ];
+		const expectedQueryLog3: any[] = expectedQueryLog2.concat({
+			type: 'sort',
+			arguments: [ sortObjectArray ],
+			normalizedArguments: [ [ sortObject, lang.mixin({ descending: false }, sortObjectArray[1]) ] ]
+		});
+		const expectedQueryLog4 = expectedQueryLog3.concat({
+			type: 'sort', arguments: [ comparator ], normalizedArguments: [ comparator ]
+		});
+		let sortedCollection: dstore.Collection<any>;
+
+		sortedCollection = store.sort(sortObject.property, sortObject.descending);
+		assert.deepEqual(sortedCollection.queryLog, expectedQueryLog1);
+
+		sortedCollection = store.sort(sortObject);
+		assert.deepEqual(sortedCollection.queryLog, expectedQueryLog2);
+
+		sortedCollection = sortedCollection.sort(sortObjectArray);
+		assert.deepEqual(sortedCollection.queryLog, expectedQueryLog3);
+
+		sortedCollection = sortedCollection.sort(comparator);
+		assert.deepEqual(sortedCollection.queryLog, expectedQueryLog4);
+	},
+
+	'restore'() {
+		const store = new ConcreteStore({
+			Model: NonDeclareBasedModel
+		});
+		const restoredObject = (<any> store)._restore({ foo: 'original' });
+		assert.strictEqual(restoredObject.foo, 'original');
+		assert.strictEqual(restoredObject.restored, true);
+		assert.isTrue(restoredObject instanceof NonDeclareBasedModel);
+	},
+
+	events() {
+		const methodCalls: string[] = [],
+			events: string[] = [];
+
+		// rely on autoEventEmits
+		const store = new  ConcreteStore();
+		function countMethodCall(method: string, object: any) {
+			methodCalls.push(method);
+			return object;
+		}
+		store.put = countMethodCall.bind(null, 'put');
+		store.add = countMethodCall.bind(null, 'add');
+		store.remove = countMethodCall.bind(null, 'remove');
+
+		store.on('add', function (event: EventObject) {
+			events.push(event.type);
+		});
+		// test comma delimited as well
+		store.on('delete', function (event: EventObject) {
+			events.push(event.type);
+		});
+
+		store.on('update', function (event: EventObject) {
+			events.push(event.type);
+		});
+
+		store.put({});
+		store.add({});
+		store.remove(1);
+
+		assert.deepEqual(methodCalls, ['put', 'add', 'remove']);
+		assert.deepEqual.bind(this, events, ['update', 'add', 'delete']);
```

Comment:
No reason, and it was hiding the fact that this test was actually broken. I've removed the bind call and fixed the test.